### ISSUE TITLE
useStorage should return defaults, when value was reset

### ISSUE THREAD

**Issue by realainov** [vueuse/vueuse#3778]
### Describe the bug

const someValue = useStorage<Record<string, number>>('some-value', {})
someValue.value = null
console.error(someValue.value) // expect {}, but I see null

RemovableRef is define that the value shouldn't be null.

### Reproduction

https://playcode.io/1754130

### System Info

```Shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M1
    Memory: 108.14 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.1 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 9.6.7 - /usr/local/bin/npm
    pnpm: 8.11.0 - ~/Library/pnpm/pnpm
    bun: 1.0.14 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 121.1.62.156
    Chrome: 121.0.6167.139
    Safari: 17.2.1
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by zekunjin** [comment]
If the value of reactive storage is set to null and then the default value is returned, it will never be null again. It seems that adding a `reset` method might be a better way.
And this would lead to a breaking change in the `useStorage` API, which might require a more in-depth discussion.

**Comment by realainov** [comment]
The value can't be null at all, look at types.
declare function useStorage<T = unknown>(key: string, defaults: MaybeRefOrGetter<null>, storage?: StorageLike, options?: UseStorageOptions<T>): RemovableRef<T>;
type RemovableRef<T> = Omit<Ref<T>, 'value'> & {
    get value(): T;
    set value(value: T | null | undefined);
};

When i call this hook like this:
const someValue = useStorage<Record<string, boolean>>('some-value', {})
It means that getter of removable ref always extends Record<string, boolean>, it cant' be null.
I can set null to reset storage, but getter can't return null.

**Comment by zekunjin** [comment]
You're right. At least the type declaration for `get` should be the same as the value for `set` in `ComputedRef`.

**Comment by realainov** [comment]
I think not, after all. I'll explain why. Then the default value becomes meaningless, the current RemovableRef type is set so that we can clean up the local storage not through a separate method, but simply by writing to the null value. At the same time, there will be no inconvenience when working with ref, checking the value for emptiness every time.

**Comment by zekunjin** [comment]
But if local storage is cleared by setting ref to `null`, and get default value as return, then it would look something like this:

```ts
const ref = useStorage<Record<string, boolean>>('KEY', {})
ref.value = null
console.log(ref.value) // Output: {}
```

Personally, It doesn't seem to align with common sense.
In current API, rather than referring to it as a default value, it feels like naming it as an init value would be more appropriate.

And if we clean up the local storage with writing a `null` value, and get `null` as return, then the `RemovableRef` value could potentially return `null`. If not check the value for emptiness, there could be a type safety issue.

```ts
const ref = useStorage<Record<string, boolean>>('KEY', {})
ref.value = null
console.log(ref.value) // Output: null
// Expect: undefined
console.log(ref.value.a) // Error: Cannot read properties of null (reading 'a')
```

Therefore, maybe modifying the type declaration of the `RemovableRef` get method and revising the description of the default value in the documentation would be a better way.

**Comment by realainov** [comment]
Let @antfu answer what he wanted to express with this construction, it will be more constructive.

**Comment by Jckpt** [comment]
+1
I would really want to have some type of reset method, instead of removing localstorage key entirely :^ )
